kobo/apps/hook/tasks.py

```diff
@@ -46,4 +55,104 @@ def retry_all_task(hooklogs_ids):
+        # So, we need to update it manually.
+        # see: http://docs.celeryproject.org/en/latest/userguide/configuration.html#beat-sync-every
+        PeriodicTask.objects.filter(task=beat_schedule.get("task")).update(last_run_at=timezone.now())
+
+        records = {}
+        max_length = 0
+
+        # Prepare data for templates.
+        # All logs will be grouped under their respective asset and user.
+        for record in queryset:
+            # if user doesn't exist in dict, add him
+            if record.hook.asset.owner.id not in records:
+                records[record.hook.asset.owner.id] = {
+                    "username": record.hook.asset.owner.username,
+                    # language is in implemented yet.
+                    # TODO add language to user table in registration process
+                    "language": getattr(record.hook.asset.owner, "language", "en"),
+                    "email": record.hook.asset.owner.email,
+                    "assets": {}
+                }
+
+            # if asset doesn't exist in user's asset dict, add it
+            if record.hook.asset.id not in records[record.hook.asset.owner.id]["assets"]:
+                max_length = 0
+                records[record.hook.asset.owner.id]["assets"][record.hook.asset.id] = {
+                    "name": record.hook.asset.name,
+                    "max_length": 0,
+                    "logs": []
+                }
+
+            # Add log to corresponding asset and user
+            records[record.hook.asset.owner.id]["assets"][record.hook.asset.id]["logs"].append({
+                "hook_name": record.hook.name,
+                "uid": record.uid,
+                "date_modified": record.date_modified,
+                "status_code": record.status_code,
+                "message": record.message
+            })
+            hook_name_length = len(record.hook.name)
+
+            # Max Length is used for plain text template. To display fixed size columns.
+            max_length = max_length if max_length > hook_name_length else hook_name_length
+            records[record.hook.asset.owner.id]["assets"][record.hook.asset.id]["max_length"] = max_length
+
+        # Get templates
+        plain_text_template = get_template("reports/failures_email_body.txt")
+        html_template = get_template("reports/failures_email_body.html")
+        email_messages = []
+
+        for owner_id, record in records.items():
+            variables = {
+                "username": record.get("username"),
+                "assets": record.get("assets")
+            }
+            # Localize templates
+            translation.activate(record.get("language"))
+            text_content = plain_text_template.render(Context(variables))
+            html_content = html_template.render(Context(variables))
+
+            msg = EmailMultiAlternatives(translation.ugettext("Hooks failures report"), text_content,
```

Comment:
Since the UI doesn't use the term "hooks," let's change this to "REST Services Failure Report"